tests/unit/pubsublite/internal/wire/partition_count_watcher_impl_test.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import asyncio
+import queue
+import threading
+from asynctest.mock import MagicMock
+import pytest
+
+from google.cloud.pubsublite import AdminClientInterface
+from google.cloud.pubsublite.internal.wire.partition_count_watcher_impl import (
+    PartitionCountWatcherImpl,
+)
+from google.cloud.pubsublite.internal.wire.publisher import Publisher
+from google.cloud.pubsublite.testing.test_utils import Box
+from google.cloud.pubsublite.types import Partition, TopicPath, CloudZone, CloudRegion
+from google.api_core.exceptions import GoogleAPICallError
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture()
+def mock_publishers():
+    return {Partition(i): MagicMock(spec=Publisher) for i in range(10)}
+
+
+@pytest.fixture()
+def topic():
+    zone = CloudZone(region=CloudRegion("a"), zone_id="a")
+    return TopicPath(project_number=1, location=zone, name="c")
+
+
+@pytest.fixture()
+def mock_admin():
+    admin = MagicMock(spec=AdminClientInterface)
+    return admin
+
+
+@pytest.fixture()
+def watcher(mock_admin, topic):
+    box = Box()
+
+    def set_box():
+        box.val = PartitionCountWatcherImpl(mock_admin, topic, 0.001)
+
+    # Initialize publisher on another thread with a different event loop.
```

Comment:
Any asyncio class cannot be accessed from a different event loop, or it is incorrect.

Reply:
Okay, before I didn't understand this was necessarily constructed in the same event loop as the publisher.

I still kind of think the property that everything works even if you call __aenter__ from a different thread than you constructed the watcher is nice. I'm happy to remove it though if you think that would be better

Reply:
Can you add a utility to the testing helpers that does exactly this? I think this is a good idea, and we should do this mroe places.